addons/packages/vsphere-cpi/1.22.3/bundle/config/overlays/update-daemonset.yaml

```diff
@@ -38,17 +38,19 @@ spec:
           #@ for arg in struct.decode(values.vsphereCPI.cloudProviderExtraArgs):
           - --(@=arg @)=(@=struct.decode(values.vsphereCPI.cloudProviderExtraArgs)[arg]@)
           #@ end
-      #@ if values.vsphereCPI.http_proxy != "" :
       #@overlay/match by=overlay.subset({"name": "vsphere-cloud-controller-manager"})
       #@overlay/match-child-defaults missing_ok=True
       - env:
+        - name: "ENABLE_ALPHA_DUAL_STACK"
+          value: "true"
```

Comment:
Should be sourced from a data value.

Reply:
cc @tylerschultz

Reply:
@vijaykatam let me share some contexts here, we combine two changes in this PR, the first one is to fix the label/annotation stuff for daemonset overlay, another change is to set the 'ENABLE_ALPHA_DUAL_STACK' env var on the DaemonSet in order to enable our dual-stack feature

Reply:
👍 just make sure to update PR description with your comment.

Reply:
done!

Reply:
> Should be sourced from a data value.

@vijaykatam this this environment variable is toggling a feature gate that works in tandem with the `ipFamily` data value which already exists.

When this is set to false, the CPI only allows `ipFamily` to be set to `ipv4` or `ipv6` -- it will fast fail if set to a dual stack value like `ipv4,ipv6` or `ipv6,ipv4`. when it is set to true, those dual stack values are allowed.

We considered three options:

1. Expose a new data value for this feature flag; users would have to toggle the feature flag when they set `ipFamily` to a dual stack value
2. Within the package, automatically but conditionally set this flag to `true` if `ipFamily` is set to a dual stack option
3. Just enable it

We thought option (3) made the most sense in this scenario.

Reply:
I would have chosen option(2) because that allows you to turn it off(for any reason) without requiring an overlay in the package install secret which is not very intuitive to use but gonna defer to your team.

Reply:
@nicolehanjing , @tylerschultz and I discussed briefly.

@nicolehanjing will remove this change for now and @tylerschultz and our team will follow up separately with another PR later.